src/yaml/parser.cr

```diff
@@ -83,7 +83,11 @@ class YAML::Parser
         key = parse_node
         @pull_parser.read_next
         value = parse_node
-        mapping[key] = value
+        if key == "<<"
+          raise "Expected value to be a Hash" unless value.is_a?(Hash)
+          mapping.merge!(value)
+        else mapping[key] = value
```

Comment:
Let's add a newline here, that always looks like an empty `elsif` on first glance to me.